lib/retailer/application/add_product_form_notifier.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:io';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:deall/core/application/product/product.dart';
+import 'package:deall/core/application/value_validator.dart';
+import 'package:deall/core/infrastructure/image_picking_repository.dart';
+import 'package:deall/retailer/application/add_product_form_state.dart';
+import 'package:deall/retailer/infrastructure/product_repository.dart';
+
+
+class AddProductFormNotifier extends StateNotifier<AddProductFormState> {
+  final ProductListRepository _productRepository;
+  final ImagePickingRepository _imagePickingRepository;
+
+  AddProductFormNotifier(this._productRepository, this._imagePickingRepository) : super(AddProductFormState.initial());
+
+  void prodNameChanged(String name){
+    state = state.copyWith(name: name);
+  }
+
+  void prodUsualPriceChanged(double usualPrice){
+    state = state.copyWith(usualPrice: usualPrice);
+  }
+
+  void prodDiscountPriceChanged(double discountedPrice){
+    state = state.copyWith(discountedPrice: discountedPrice);
+  }
+
+  void prodImageChanged(File? imageFile){
+    state = state.copyWith(imageFile: imageFile);
+  }
+
+  void prodDescriptionChanged(String description){
+    state = state.copyWith(description: description);
+  }
+
+  void prodAvailabilityChanged(){
+    state = state.copyWith(availability: !state.availability);
+  }
+
+  void _validateInputs(){
+    var stateCopy = state.copyWith(showErrorMessage: true);
+    //validate name
+    final nameValidate = validateNotEmpty(state.name);
+    nameValidate.fold(
+      (valueFailure) => valueFailure.maybeWhen(
+        empty: () {
+          stateCopy = stateCopy.copyWith(
+            nameErrorMessage: 'Product name cannot be empty',
+          );
+        },
+        orElse: () {},
+      ),
+      (r) => stateCopy = stateCopy.copyWith(nameErrorMessage: null),
+    );
+    //validate name
+    final usualPriceValidate = validateNotEmpty(state.name);
+    usualPriceValidate.fold(
+      (valueFailure) => valueFailure.maybeWhen(
+        empty: () {
+          stateCopy = stateCopy.copyWith(
+            usualPriceErrorMessage: 'Usual price of product cannot be empty',
+          );
+        },
+        orElse: () {},
+      ),
+      (r) => stateCopy = stateCopy.copyWith(usualPriceErrorMessage: null),
+    );
+    state = stateCopy;
+  }
+
+  Future<void> addProduct(String uid) async {
+    _validateInputs();
+
+    //if the input are valid
+    if (state.nameErrorMessage == null &&
+        state.usualPriceErrorMessage == null) {
+      state = state.copyWith(
+        isSaving: true,
+        hasConnection: true,
+      );
+
+      final newProduct = Product(
+        id: '',
+        name: state.name,
+        usualPrice: state.usualPrice,
+        discountedPrice: state.discountedPrice,
+        image: state.imageFile.toString(),
```

Comment:
this shouldn't be working need to use uploadProductImage in imagePicking repo, refer to retailer sign up notifier.